okhttp/src/main/java/okhttp3/HttpUrl.java

```diff
@@ -772,6 +772,23 @@ public Builder addPathSegment(String pathSegment) {
       return this;
     }
 
+    public Builder addPathSegments(String groupedPathSegments) {
+      if (groupedPathSegments == null) {
+        throw new IllegalArgumentException("groupedPathSegments == null");
+      }
+      int startIndex = 0;
+      int endIndex = groupedPathSegments.indexOf('/');
+      while (endIndex != -1) {
```

Comment:
With `Util.delimiterOffset()` it should be straightforward to call `push()` only once within the loop. Something like

```
int offset = 0;
do {
  int segmentEnd = delimiterOffset(...)
  push(groupedPathSegments, ...);
  offset = segmentEnd + 1;
} while (offset <= groupedPathSegments.length());
```
I think one interesting test case will be when the input ends with `/` and I think the resulting URL should then end with `/`. We should probably also confirm that the code works if the input starts with `/`.